Mathlib/Algebra/Order/Floor/Div.lean

```diff
@@ -0,0 +1,210 @@
+/-
+Copyright (c) 2023 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies
+-/
+import Mathlib.Algebra.Order.Module.Defs
+import Mathlib.Data.Finsupp.Order
+import Mathlib.Data.Nat.Order.Basic
+import Mathlib.Order.GaloisConnection
+
+/-!
+# Flooring, ceiling division
+
+This file defines division rounded up and down.
+
+The setup is an ordered monoid `α` acting on an ordered monoid `β`. If `a : α`, `b : β`, we would
+like to be able to "divide" `b` by `a`, namely find `c : β` such that `a • c = b`.
+This is of course not always possible, but in some cases at least there is a least `c` such that
+`b ≤ a • c` and a greatest `c` such that `a • c ≤ b`. We call the first one the "ceiling division
+of `b` by `a`" and the second one the "flooring division of `b` by `a`"
+
+If `α` and `β` are both `ℕ`, then one can check that our flooring and ceiling divisions really are
+the floor and ceil of the exact division.
+If `α` is `ℕ` and `β` is the functions `ι → ℕ`, then the flooring and ceiling divisions are taken
+pointwise.
+
+In order theory terms, those operations are respectively the right and left adjoints to the map
+`b ↦ a • b`.
+
+## Main declarations
+
+* `FloorDiv`: Typeclass for the existence of a flooring division, denoted `b ⌊/⌋ a`.
+* `CeilDiv`: Typeclass for the existence of a ceiling division, denoted `b ⌈/⌉ a`.
+
+Note in both cases we only allow dividing by positive inputs. We enforce the following junk values:
```

Comment:
I guess there are quite reasonable settings in which one has both a `FloorDiv` and a `CeilDiv`, and I expect that in that case `FloorDiv \le CeilDiv`. Could you please add a lemma about that?